sdk-cpp/src/do_download.cpp

```diff
@@ -58,23 +58,61 @@ void download::download_url_to_path(const std::string& uri, const std::string& d
     download_url_to_path(uri, downloadFilePath, isCancelled, timeOut);
 }
 
+// Ensures download is finalized/aborted in the agent to avoid stale download objects.
+// Future: We could expect callers to explicitly opt-in to persist downloads in the agent and
+//   use that as an indicator to perform/skip the cleanup directly in ~download().
```

Comment:
Does the agent currently have a self-cleanup mechanism to abort/expire jobs after a set period of time (24hr?)

Reply:
Windows dosvc does but the expiration is 30 days. ADU DO keeps it until shutdown (no persistence).

Reply:
And it's stored in memory until then? This could explain the memory usage I saw earlier (where after many downloads, our memory usage increased) - though this would only occur in the case the start threw. so maybe not